docs/advanced-features/sitemaps.md

```diff
@@ -0,0 +1,179 @@
+
+A **sitemap** defines the relationship between pages of your site. Search engines utilize
+this file to more accurately index your site. You can also provide additional information
+such as last updated time, how frequently the page changes, and more.
+
+A **robots.txt** file tells search engines which pages or files the crawler can or can't request from your site.
+
+## Static Sitemaps
+
+If your site does not update frequently, you might currently have a static sitemap.
+This is a basic `.xml` file defining the content of your site. Here's a simple example:
+
+```xml
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+  <url>
+      <loc>https://nextjs.org</loc>
+  </url>
+  <url>
+      <loc>https://nextjs.org/docs</loc>
+  </url>
+</urlset>
+```
+
+## Dynamic Sitemaps
+
+If your site frequently changes, you should dynamically create a sitemap. First, let's explore an example where your content is based on [file-system routing](/docs/routing/introduction.md) (e.g., contained inside the `/pages` directory).
+
+This example is using [globby](https://github.com/sindresorhus/globby) to fetch our list of routes.
+
+```bash
+npm install globby --save-dev
+```
+
+Next, we can create a Node script at `scripts/generate-sitemap.mjs`.
+This file will dynamically build a sitemap based on your `/pages` directory.
+
+```js
+import { writeFileSync } from 'fs'
+import { globby } from 'globby'
+
+async function generate() {
+  const pages = await globby([
+    'pages/*.js', // Include all pages
+    '!pages/_*.js', // Exclude _app or _document
+    '!pages/api', // Exclude API Routes
+    '!pages/404.js', // Exclude 404
+  ])
+
+  const sitemap = `
+    <?xml version="1.0" encoding="UTF-8"?>
+    <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+        ${pages
+          .map((page) => {
+            const path = page.replace('pages', '').replace('.js', '')
+            const route = path === '/index' ? '' : path
+
+            return `
+              <url>
+                  <loc>${`https://nextjs.org${route}`}</loc>
+              </url>
+            `
+          })
+          .join('')}
+    </urlset>
+    `
+
+  writeFileSync('public/sitemap.xml', sitemap)
+}
+
+generate()
+```
+
+> **Note:** This Node.js script is using [ES Modules](https://nodejs.org/api/esm.html), denoted by the `.mjs` extension. You could also write this script with [CommonJS](https://nodejs.org/api/modules.html#modules_modules_commonjs_modules) if you prefer.
+
+Finally, add `postbuild` script in your `package.json` to run this script after `next build` completes.
+Your generated sitemap file gets created at `public/sitemap.xml`, which is then served as a [static file](/docs/basic-features/static-file-serving.md) at the root of your site.
+
+```json
+{
+  "scripts": {
+    "build": "next build",
+    "postbuild": "node ./scripts/generate-sitemap.mjs",
```

Comment:
Is this a standard? I think they actually would need to modify the `build` script

Reply:
https://docs.npmjs.com/cli/v7/using-npm/scripts#pre--post-scripts

Reply:
FYI `pre` and `post` hooks for user-defined scripts are not supported in yarn berry 2.x \ 3.x

> In particular, we intentionally don't support arbitrary pre and post hooks for user-defined scripts (such as prestart). This behavior, inherited from npm, caused scripts to be implicit rather than explicit, obfuscating the execution flow. It also led to surprising executions with yarn serve also running yarn preserve.

See also: https://yarnpkg.com/advanced/lifecycle-scripts